DDF-3651 Coordinate Value Calculation for Map Center

Schachte opened the pull request:
#### What does this PR do?
Previously the cursor location at the time it moves off of the map was displayed as the current coordinates in the bottom left hand side of the map. 

This PR updates the lat/lon coords to represent whatever the current center view of the map is and updates the bottom left-hand div accordingly. 

#### Who is reviewing it? 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
@bdeining 
@GabrielFabian 
@Lambeaux 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@andrewkfiedler
@adimka  
#### How should this be tested? (List steps with links to updated documentation)
#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-3651](https://codice.atlassian.net/browse/DDF-3651)
#### Screenshots (if appropriate)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

GabrielFabian commented on catalog/ui/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/map.cesium.js:
```diff
@@ -161,6 +163,7 @@ module.exports = function CesiumMap(insertionElement, selectionInterface, notifi
     setupTooltip(map, selectionInterface);
 
     function updateCoordinatesTooltip(position) {
+
```
nit: extra newline here is inconsistent with your other functions.

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/map.cesium.js:
```diff
@@ -36,6 +36,8 @@ var mtgeo = require('mt-geo');
 var defaultColor = '#3c6dd5';
 var eyeOffset = new Cesium.Cartesian3(0, 0, 0);
 var pixelOffset = new Cesium.Cartesian2(0.0, 0);
+var viewer;
```
❗️ We can't pull these variables (`viewer`, `currentMapCenter`) out as singletons, otherwise they'll be shared by all the open cesium maps.

Schachte replied:
Added openlayers support and changed a few things around here: 6d7b17e72b9753c6c118cebdf28b837ede316650

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/visualization/maps/openlayers/map.openlayers.js:
```diff
@@ -170,6 +176,17 @@ module.exports = function OpenlayersMap(insertionElement, selectionInterface, no
                 });
             });
+
+            $(map.getTargetElement()).on('mouseleave', function() {
```
The map can change positions when the mouse isn't on it (we fly to locations), we'll need to account for those scenarios as well.

Schachte replied:
@andrewkfiedler What's the desired result when a user flies to a location? They click on a location on the side-pane and the coordinates update to that location?

andrewkfiedler replied:
I think updating the location to be the map center would be fine (it'd theoretically be the location of the result).

andrewkfiedler replied:
There are other scenarios though as well, such as using the gazetteer to fly to a location.

Schachte replied:
Adjustments based on our conversation addressed in db5a713

Schachte commented:
@andrewkfiedler Ok, based on our discussion I removed the center map logic and updated coordinate box to show the "Cursor Off Map" label when the cursor is off the map.

andrewkfiedler commented on catalog/ui/catalog-ui-search/src/main/webapp/component/map-info/map-info.less:
```diff
@@ -1,19 +1,28 @@
+@import url('https://fonts.googleapis.com/css?family=Inconsolata');
```
We shouldn't be pulling in a font here.

Schachte replied:
fixed db5a713

Schachte commented:
build now

Schachte commented:
build now

Schachte commented:
build now (Failed due to network error)

Schachte commented:
Address design changes: https://jira.di2e.net/secure/attachment/159275/159275_coorupdate.mov

bdeining commented on ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/openlayers/map.openlayers.js:
```diff
@@ -170,6 +172,14 @@ module.exports = function OpenlayersMap(insertionElement, selectionInterface, no
             });
+
+            $(map.getTargetElement()).on('mouseleave', function() {
+                $( ".mapInfo" ).hide();
```
✏️ extra whitespace?

bdeining commented on ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/map.cesium.js:
```diff
@@ -266,6 +271,14 @@ module.exports = function CesiumMap(insertionElement, selectionInterface, notifi
             });
+
+            $(map.scene.canvas).on('mouseleave', function() {
+                $(".mapInfo" ).hide();
```
✏️ extra whitespace?

bdeining commented on ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/map.cesium.js:
```diff
@@ -164,11 +164,16 @@ module.exports = function CesiumMap(insertionElement, selectionInterface, notifi
         if (map.scene.pickPositionSupported) {
             var cartesian = map.scene.pickPosition(position);
             if (Cesium.defined(cartesian)){
+                $(".mapInfo" ).show();
```
✏️ extra whitespace?

bdeining commented on ui/packages/catalog-ui-search/src/main/webapp/component/map-info/map-info.less:
```diff
@@ -5,15 +5,22 @@
     position: absolute;
-    left: 0px;
-    bottom: 0px;
+    left: 10px;
```
should this be relative  to `@minimumSpacing`?

bdeining commented on ui/packages/catalog-ui-search/src/main/webapp/component/map-info/map-info.less:
```diff
@@ -5,15 +5,22 @@
     text-align: left;
     padding: @minimumSpacing;
     max-width: 50%;
+    border: 1px solid transparent;
```
@andrewkfiedler should we use css rules here that are already defined?

bdeining commented:
build now
